app/uk/gov/hmrc/agentsubscription/connectors/AgentOverseasApplicationConnector.scala

```diff
@@ -61,15 +62,21 @@ class AgentOverseasApplicationConnector @Inject() (
       http.GET(url.toString).map { response =>
         val json = response.json.head
         val status = (json \ "status" \ "typeIdentifier").as[String]
-        val safeId = (json \ "safeId").asOpt[SafeId]
+
+        val safeId = (json \ "safeId").validateOpt[SafeId] match {
```

Comment:
if this is to throw an exception, you might as well force it and use validate[SafeId]

Reply:
I think it's ok. It's an optional field, so if it's valid for it not to be present and in such a case there's nothing to validate and the value should be a `None`. However if the field is present, then it's value should be validated.